docs/source/examples/word-count.rst

```diff
@@ -0,0 +1,247 @@
+namenode.
+
+.. code-block:: python
+
+   >>> hdfs = hdfs3.HDFileSystem('NAMENODE_HOSTNAME', port=NAMENODE_PORT)
+
+Initalize a connection to the ``distributed`` executor, replacing
+``EXECUTOR_IP`` and ``EXECUTOR_PORT`` with the hostname and port of the
+``distributed`` scheduler.
+
+.. code-block:: python
+
+   >>> e = Executor('EXECUTOR_IP:EXECUTOR_PORT')
+
+Generate a list of filenames from the text data in HDFS:
+
+.. code-block:: python
+
+   >>> filenames = hdfs.glob('/tmp/enron/*/*')
+   >>> print(filenames[:5])
+
+   ['/tmp/enron/edrm-enron-v2_nemec-g_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_ring-r_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_bailey-s_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_fischer-m_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_geaccone-t_xml.zip/merged.txt']
+
+Print the first 1024 bytes of the first text file:
+
+.. code-block:: python
+
+   >>> print(hdfs.head(filenames[0]))
+
+   b'Date: Wed, 29 Nov 2000 09:33:00 -0800 (PST)\r\nFrom: Xochitl-Alexis Velasc
+   o\r\nTo: Mark Knippa, Mike D Smith, Gerald Nemec, Dave S Laipple, Bo Barnwel
+   l\r\nCc: Melissa Jones, Iris Waser, Pat Radford, Bonnie Shumaker\r\nSubject:
+    Finalize ECS/EES Master Agreement\r\nX-SDOC: 161476\r\nX-ZLID: zl-edrm-enro
+   n-v2-nemec-g-2802.eml\r\n\r\nPlease plan to attend a meeting to finalize the
+    ECS/EES  Master Agreement \r\ntomorrow 11/30/00 at 1:30 pm CST.\r\n\r\nI wi
+   ll email everyone tomorrow with location.\r\n\r\nDave-I will also email you 
+   the call in number tomorrow.\r\n\r\nThanks\r\nXochitl\r\n\r\n***********\r\n
+   EDRM Enron Email Data Set has been produced in EML, PST and NSF format by ZL
+    Technologies, Inc. This Data Set is licensed under a Creative Commons Attri
+   bution 3.0 United States License <http://creativecommons.org/licenses/by/3.0
+   /us/> . To provide attribution, please cite to "ZL Technologies, Inc. (http:
+   //www.zlti.com)."\r\n***********\r\nDate: Wed, 29 Nov 2000 09:40:00 -0800 (P
+   ST)\r\nFrom: Jill T Zivley\r\nTo: Robert Cook, Robert Crockett, John Handley
+   , Shawna'
+
+Create a function to count words in each file:
+
+.. code-block:: python
+
+   >>> def count_words(fn):
+   ...     word_counts = defaultdict(int)
+   ...     with hdfs.open(fn) as f:
+   ...         for line in f:
+   ...             for word in line.split():
+   ...                 word_counts[word] += 1
+   ...     return word_counts
+
+Count the number of words in the first text file locally:
+
+.. code-block:: python
+
+   >>> counts = count_words(filenames[0])
+   >>> print(sorted(counts.items(), key=lambda k_v: k_v[1], reverse=True)[:10])
+
+   [(b'the', 144873),
+    (b'of', 98122),
+    (b'to', 97202),
+    (b'and', 90575),
+    (b'or', 60305),
+    (b'in', 53869),
+    (b'a', 43300),
+    (b'any', 31632),
+    (b'by', 31515),
+    (b'is', 30055)]
+
+Count the number of words in the first text file on a ``distributed`` worker:
```

Comment:
For me, it was helpful to build up a mental model of 1) local, 2) `e.submit`, 3) `e.map` and working with a single future vs. several futures. I added some more descriptive text of why we are doing this.